lib/sync/default-dataHandlers.js

```diff
@@ -43,7 +53,7 @@ module.exports = function (db) {
                      'doRead : ' + dataset_id +
                       ' :: ' + uid +
                       ' :: meta=' + util.inspect(meta_data));
-      mongo.collection(dataset_id).findOne({"_id" : ObjectID(uid)}, function(err, found){
+      mongo.collection(dataset_id).findOne({"_id" : convertToObjectId(dataset_id, uid)}, function(err, found){
```

Comment:
I think we just need to make sure `found` is not null if we are doing anything with it. Otherwise we can just pass it to the caller and let the caller decide what to do.